### ISSUE TITLE
Question about StableDiffusionInpaintPipelineLegacy and StableDiffusionInpaintPipeline

### ISSUE THREAD

**Issue by jinwonkim93** [huggingface/diffusers#3824]
Did you not use original latent in StableDiffusionInpaintPipeline intentionally? If yes I think it is hard to say StableDiffusionInpaintPipeline can achieve exactly the same functionality as StableDiffusionInpaintPipelineLegacy. What do think?

https://github.com/huggingface/diffusers/blob/f7cc9adc05c36985c8b3f3e95a54940ddac5b4ca/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint_legacy.py#L140

**Comment by patrickvonplaten** [comment]
Actually `StableDiffusionInpaintPipeline` should achieve exactly the same functionality. Could you post an example that you cannot do with `StableDiffusionInpaintPipeline`?

**Comment by jinwonkim93** [comment]
I thought StableDiffusionInpaintLegacyPipeline does not change unmasked area, but it did. i have tested both pipelines, the only difference was loading stable diffusion weight (4 dim) but StablediffusionInpaintPipeline can't . But this is not my usecase so i will close this issue. Thank you.

**Comment by patrickvonplaten** [comment]
Hey @jinwonkim93,

Note that `StableDiffusionInpaintPipeline` can load stable diffusion weights (4 channels dims). E.g. the following works just fine:

```py
import PIL
import requests
import torch
from io import BytesIO

from diffusers import StableDiffusionInpaintPipeline


def download_image(url):
    response = requests.get(url)
    return PIL.Image.open(BytesIO(response.content)).convert("RGB")


img_url = "https://raw.githubusercontent.com/CompVis/latent-diffusion/main/data/inpainting_examples/overture-creations-5sI6fQgYIuo.png"
mask_url = "https://raw.githubusercontent.com/CompVis/latent-diffusion/main/data/inpainting_examples/overture-creations-5sI6fQgYIuo_mask.png"

init_image = download_image(img_url).resize((512, 512))
mask_image = download_image(mask_url).resize((512, 512))

pipe = StableDiffusionInpaintPipeline.from_pretrained(
    "runwayml/stable-diffusion-v1-5", torch_dtype=torch.float16
)
pipe = pipe.to("cuda")

prompt = "Face of a yellow cat, high resolution, sitting on a park bench"
image = pipe(prompt=prompt, image=init_image, mask_image=mask_image).images[0]
```

**Comment by jinwonkim93** [comment]
@patrickvonplaten Thank you!
